crates/rover-client/src/operations/contract/publish/types.rs

```diff
@@ -0,0 +1,40 @@
+use crate::{operations::contract::publish::runner::contract_publish_mutation, shared::GraphRef};
+
+pub(crate) type MutationContractVariantUpsertResult =
+    contract_publish_mutation::ContractPublishMutationGraphUpsertContractVariant;
+pub(crate) type MutationResponseData = contract_publish_mutation::ResponseData;
+
+type MutationVariables = contract_publish_mutation::Variables;
+
+use serde::Serialize;
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct ContractPublishInput {
+    pub graph_ref: GraphRef,
+    pub source_variant: Option<String>,
```

Comment:
this should probably just be a `String` since we require this as an argument input. if the API says it's nullable, just wrap it up in `Some()` since we will always send it. then we can remove some of those adhoc runtime errors

Reply:
So the `--source-variant` argument is actually an optional one in the command (in addition to being optional in the API).

We made this argument optional since the source variant can't be changed after contract creation. So basically folks will specify it once during contract creation, and won't need to specify it again. If they do provide it again, it must match what they specified during creation, or the backend will return an error.

Reply:
ooh got it! ok 👍🏼 thanks